src/Keboola/DbExtractor/Configuration/BaseExtractorConfigDefinition.php

```diff
@@ -4,13 +4,22 @@
 
 namespace Keboola\DbExtractor\Configuration;
 
+use Keboola\Component\Config\BaseConfigDefinition;
 use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
 use Symfony\Component\Config\Definition\Builder\TreeBuilder;
-use Symfony\Component\Config\Definition\ConfigurationInterface;
 
-abstract class BaseExtractorConfigDefinition implements ConfigurationInterface
+abstract class BaseExtractorConfigDefinition extends BaseConfigDefinition
 {
-    abstract public function getConfigTreeBuilder(): TreeBuilder;
+    protected function getRootDefinition(TreeBuilder $treeBuilder): ArrayNodeDefinition
+    {
+        $node = parent::getRootDefinition($treeBuilder);
+
+        $node
```

Comment:
Normálně nevyžadujeme atributy na root úrovni, ale tím, že komponenta tu "action" potřebuje, tak mi to přijde správně. 

Obecně by jinak stačilo to, že je v configu `ignoreExtraKeys(false)`

Reply:
Takze je to ok? Uplne jsem to nepochopil, co si tim chtel rict.

Reply:
Aha, sorry. Ano, je to OK (protože bez toho ta appka nepoběží). 

Jen poznámka propřípad že budeš ještě někdy zas psát nějakou komponentu, přečteš si, co všechno je v config.json a chtěl bys to všechno zadat do té definice:

Při tvorbě php-component dohodli, že nebudeme dělat ty věci na prvním levelu vůbec vyžadovat v config definition, protože by to zaseklo zpětnou kompatiblitu. 

Dejme tomu, že bysme chtěli zrušit "action" nebo ji udělat nepovinnou. Tak pokud by to bylo v config definition všude, tak by to nešlo, protože by na tom padaly i konfigurace, které "action" vůbec nepoužívají, jen ji mají v ConfigDefinition "pro jistotu". 

Je to takhle jasnější? Nebo jsem se zas moc rozepsal?

Reply:
Jo jasne, uz je to jasnejsi. Tady je ten `action` jako optional, takze to zadny BC break nezavede. Nicmene kdyz na to tak koukam unika mi, proc jsme to nadefinovali takto, kdyz `ignoreExtraKeys(false)` uz v prekovi je. Jeste to cele overim, jestli je to vubec nutne.